ts/screens/wallet/__tests__/AddCardScreen.test.tsx

```diff
@@ -0,0 +1,126 @@
+import { none, some } from "fp-ts/lib/Option";
+import * as React from "react";
+import { createStore } from "redux";
+import { fireEvent } from "@testing-library/react-native";
+import I18n from "../../../i18n";
+import ROUTES from "../../../navigation/routes";
+import { applicationChangeState } from "../../../store/actions/application";
+import { appReducer } from "../../../store/reducers";
+import { GlobalState } from "../../../store/reducers/types";
+import { renderScreenFakeNavRedux } from "../../../utils/testWrapper";
+import AddCardScreen, { NavigationParams } from "../AddCardScreen";
+import { isValidCardHolder } from "../../../utils/input";
+
+const mockPresentFn = jest.fn();
+jest.mock("../../../utils/bottomSheet", () => ({
+  __esModule: true,
+  useIOBottomSheet: () => ({ present: mockPresentFn })
+}));
+
+jest.unmock("react-navigation");
+jest.mock("react-native-share", () => ({
+  open: jest.fn()
+}));
+
+const aValidCardHolder = "Mario Rossi";
+const aValidPan = "1234 5678 9123 4568";
+const aValidExpirationDate = "12/99";
+const aValidSecurityCode = "123";
+
+const anInvalidCardHolder = "Màriò Ròssì";
+describe("AddCardScreen", () => {
+  beforeEach(() => jest.useFakeTimers());
+  it("should show the continue button blocked if the aren't data", () => {
+    const component = getComponent();
+    const continueButton = component.queryByText(
+      I18n.t("global.buttons.continue")
+    );
+    expect(continueButton).not.toBeNull();
+    expect(continueButton).toBeDisabled();
+  });
+
+  it("should show the continue button active if all the field are filled correctly", () => {
+    const component = getComponent();
+    const cardHolderInput = component.queryByTestId("cardHolderInput");
+    const panInputMask = component.queryByTestId("panInputMask");
+    const expirationDateInput = component.queryByTestId(
+      "expirationDateInputMask"
+    );
+    const securityCodeInput = component.queryByTestId("securityCodeInputMask");
+    const continueButton = component.queryByText(
+      I18n.t("global.buttons.continue")
+    );
+
+    expect(cardHolderInput).not.toBeNull();
+    expect(panInputMask).not.toBeNull();
+    expect(expirationDateInput).not.toBeNull();
+    expect(securityCodeInput).not.toBeNull();
+
+    if (
+      cardHolderInput &&
+      panInputMask &&
+      expirationDateInput &&
+      securityCodeInput
+    ) {
+      fireEvent.changeText(panInputMask, aValidPan);
+      fireEvent.changeText(expirationDateInput, aValidExpirationDate);
+      fireEvent.changeText(securityCodeInput, aValidSecurityCode);
+      fireEvent.changeText(cardHolderInput, aValidCardHolder);
+    }
+
+    expect(continueButton).not.toBeDisabled();
+  });
+  it("should show the continue button blocked if all the cardHolder is invalid", () => {
```

Comment:
```suggestion
  it("should show the continue button disabled if the cardHolder is invalid", () => {
```

Reply:
Solved in [a2a0608](https://github.com/pagopa/io-app/pull/2833/commits/a2a06081fc62f586e61abb5c0f40559b97c07b0c)